docs/2.7.0/docs/daml-finance/tutorials/instrument-extension-packages/bond-extension.rst

```diff
@@ -271,3 +271,20 @@ How do I redeem a bond?
 
 On the redemption date, both the last coupon and the redemption amount will be paid. This is
 processed in the same way as a single coupon payment described above.
+
+How do I view the terms of a bond instrument?
+=============================================
+
+There are several ways to access the data of a contract:
+
+#. :ref:`fetch <daml-ref-fetch>` the contract using its contract ID. This requires the submitting
+   party to be a stakeholder of the contract.
+#. :ref:`fetchByKey <daml-ref-fetch-by-key>`: Similar to ``fetch``, but using a
```

Comment:
We don't use contract keys for bond instruments, so this would not work.

We use "interface keys" + the reference work-around + the `fetchInstrumentByKey` utility function hiding the work-around

Reply:
I don't find a reference to `fetchInstrumentByKey` in the code base. Also, I wonder whether we should mention that approach or if it is better to just compare `fetch` vs `GetView`.

Reply:
You're right, it's called `fetchInterfaceByKey` and needs to be used as follows:

`  claimInstrument <- fetchInterfaceByKey @BaseInstrument.R instrument`

I feel it is easier to just compare `fetch` vs `getView` without delving into the interface keys work-around. I wonder where it's a good place to store this pattern that we use (together with e.g. the factory pattern)

Reply:
I opened a separate issue for the Factory pattern: https://github.com/digital-asset/daml-finance/issues/850